### PR TITLE
Added waitForVisibleElement Function

### PR DESCRIPTION
**Pull request by pshakya**

### CODE REVIEW

**Comment by grawk** [comment]
Hi @pshakya thanks for your contribution. Few things:

1. please change name to waitForElementVisible. If we keep adding more of these the functions will look more consistent if the "adjective" is last. i.e. waitForElementVisible, waitForElementHidden, etc.
2. please add documentation to README
3. please add a positive and negative test case here: https://github.com/paypal/nemo-drivex/blob/develop/test/plugin.js

You can use this application for any UI validations: 
https://warm-river-3624.herokuapp.com/waits

Here is the relevant HTML from that page:

```html
    <h1>Sample latency stuff</h1>
    <form>
        <input type="button" class="wait" value="do something in 4s"/>
    <div class="hide" id="outy">Show me!!</div>
</form>
```

* Negative case, without clicking the button, `waitForElementVisible` throws an error
* Positive case, with clicking, `waitForElementVisible` returns true

File: test/plugin.js [hunk 1]
```diff
@@ -144,6 +144,32 @@ describe("nemo-drivex @plugin@", function () {
...
+   * @param timeout {Number}
+   * @param msg {String} optional message for any error messages
+   * @returns {Promise} resolves to true or false
+   */
+  describe("@waitForElementVisible@ method", function() {
+    it("should be visible", function () {
+      assert.ok(nemo.drivex.waitForElementVisible);
+    });
+    it("should return true when an element is visible", function(done) {
+      console.log(nemo.wd.By.tagName);
```
Comment (grawk) [inline test/plugin.js]: Please remove this line

File: lib/drivex.js [hunk 2]
```diff
@@ -95,6 +95,27 @@ DriveX.prototype = {
...
+   * @returns {Promise} resolves to true or throw error
+   */
+  waitForElementVisible: function (locator, timeout, msg) {
+    var self = this;
+    log('waitForElementVisible', locator);
+    return this.driver.wait(function() {
+      return self.visible(locator);
+    }, timeout, msg).then(function() {
+      log('waitForElementVisible: found', locator);
+      return true;
```
Comment (nikulkarni) [inline lib/drivex.js]: you should check the value passed to the callback before returning true.

```
return self.visible(locator);
   }, timeout, msg).then(function(isVisible) {
      log('waitForElementVisible: '+ isVisible, locator);
      return isVisible;
}
```

Comment (grawk) [inline lib/drivex.js]: Oh good catch.

File: test/plugin.js [hunk 3]
```diff
@@ -144,6 +144,31 @@ describe("nemo-drivex @plugin@", function () {
     });
   });
   /**
+   * waitForElementVisible Wait for timeout milliseconds for the WebElement to be visible
+   * @param locator {LocatorJSON}
+   * @param timeout {Number}
+   * @param msg {String} optional message for any error messages
+   * @returns {Promise} resolves to true or false
+   */
+  describe("@waitForElementVisible@ method", function() {
```
Comment (grawk) [inline test/plugin.js]: I think it'd be good to make the positive test case verify that there is latency while waiting for a DOM element to be visible. You can use this test as a guide:
https://github.com/paypal/nemo-view/blob/master/test/methods.js#L80-L98

**Pull request merged**